torchinductor/lowering.py

```diff
@@ -430,12 +436,13 @@ def where(cond, a, b):
     def fn(*args):
         return ops.where(*args)
 
-    if isinstance(a, (float, int)):
-        a = constant_like(a)(b)
-    if isinstance(b, (float, int)):
-        b = constant_like(b)(a)
-
-    dtype = torch.promote_types(a.get_dtype(), b.get_dtype())
+    a, b = promote_constants([a, b])
+    # if isinstance(a, (float, int)):
+    #     a = constant_like(a)(b)
+    # if isinstance(b, (float, int)):
+    #     b = constant_like(b)(a)
+    # dtype = torch.promote_types(a.get_dtype(), b.get_dtype())
+    dtype = a.get_dtype()
```

Comment:
This line causes a confusing error for some reason, need to debug.

Reply:
What's the error?

Reply:
<img width="975" alt="image" src="https://user-images.githubusercontent.com/6355099/195223988-a568c4c8-ccd3-47b2-9674-0b14acdaa7ff.png">

Reply:
Found the issue - the python dispatcher throws this error if you hit a aten op with no tensor inputs.